var/spack/repos/builtin/packages/clingo/package.py

```diff
@@ -36,21 +37,26 @@ class Clingo(CMakePackage):
     variant("docs", default=False, description="build documentation with Doxygen")
     variant("python", default=True, description="build with python bindings")
 
+    # See https://github.com/potassco/clingo/blob/v5.5.2/INSTALL.md
     depends_on('cmake@3.1:', type='build')
+    depends_on('cmake@3.18:', type='build', when='@5.5:')
 
     depends_on('doxygen', type="build", when="+docs")
-    depends_on('re2c@0.13:', type="build")
-    depends_on('bison@2.5:', type="build")
-
-    depends_on('python', type=("build", "link", "run"), when="+python")
-    extends('python', when='+python')
-
-    # Clingo 5.5.0 supports Python 3.6 or later and needs CFFI
-    depends_on(
-        'python@3.6.0:',
-        type=('build', 'link', 'run'), when='@5.5.0: +python'
-    )
-    depends_on('py-cffi', type=('build', 'run'), when='@5.5.0: +python')
+
+    with when('@spack'):
+        depends_on('re2c@0.13:', type="build")
+        depends_on('bison@2.5:', type="build")
+
+    with when('@master'):
```

Comment:
I *think* you can use `with when('@spack,master')` here, which I only bring up because the dependencies are identical.

Reply:
I usually find not using commas more readable, but I was honestly in doubt here. Since you mention it, I made the modification.